### PR TITLE
Vcb 26 input buttons

### PR DESCRIPTION
**Pull request by stefenz**

### CODE REVIEW

**Review by PhilSchni — APPROVED** [review]

File: test/board/devices/input/buttons.test.ts [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+})
+
+describe('test release() function', () => {
+  test('test passing invalid positions to release() function', () => {
+    expect(() => buttons.press(4)).toThrow(
+      new Error('Button position 4 does not exist.')
+    )
+    expect(() => buttons.press(-2)).toThrow(
+      new Error('Button position -2 does not exist.')
+    )
```
Comment (m-dot-wigger) [inline test/board/devices/input/buttons.test.ts:74]: copy paste ;-)
release function should be called. Perhaps use other indexes like -1 and 22 or so.

File: test/board/devices/input/buttons.test.ts [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+    expect(buttons.isPressed(0)).toBe(false)
+    expect(buttons.isPressed(1)).toBe(false)
+    expect(buttons.isPressed(2)).toBe(false)
+    expect(buttons.isPressed(3)).toBe(false)
+  })
+  test('test passing the same value to release() function again', () => {
+    buttons.release(0)
+    expect(buttons.isPressed(0)).toBe(false)
+    buttons.release(0)
+    expect(buttons.isPressed(0)).toBe(false)
```
Comment (m-dot-wigger) [inline test/board/devices/input/buttons.test.ts:91]: perhaps check if all unchanged like on line 19:
`expect(buttons.readByte(address)).toEqual(byte_0000_0101)`

**Review by m-dot-wigger — CHANGES_REQUESTED** [review]
nice placement and highlighting of button if pressed or not.

according to test coverage report line 56 of buttons.ts is not called with test for release on invalid position

File: et q [hunk 3]
```diff
@@ -0,0 +1,448 @@
+[33mcommit 22679a554842c13f66347ba31969b5abdb4b3e4a[m[33m ([m[1;36mHEAD -> [m[1;32mvcb-26-input-buttons[m[33m)[m
+Author: Silvan <niggsil2@students.zhaw.ch>
+Date:   Mon Apr 11 18:34:43 2022 +0200
+
+    add buttons memory representation
+
+[33mcommit 9bb522005a43579aa7b73b391fb58f8b52b003a0[m[33m ([m[1;33mtag: 0.232[m[33m, [m[1;31morigin/vcb-26-input-buttons[m[33m)[m
+Author: Raphael Krebs <48808406+akrosis@users.noreply.github.com>
```
[the remark this answers is not in the document]
Comment (ni507) [reply et q]: Deleted.

File: test/board/devices/input/buttons.test.ts (hunk 1 above)
Comment (ni507) [reply test/board/devices/input/buttons.test.ts]: Changed.

File: test/board/devices/input/buttons.test.ts (hunk 2 above)
Comment (ni507) [reply test/board/devices/input/buttons.test.ts:91]: Added.

Comment (m-dot-wigger) [reply test/board/devices/input/buttons.test.ts:91]: had to fix the expected value

**Review by m-dot-wigger — APPROVED** [review]
looks all good to me

**Pull request merged**